doc/changes/v2.1.0.rst

```diff
@@ -34,6 +34,27 @@ NCCL is now fetched from PyPI
 
 In the previous version, XGBoost statically linked NCCL, which significantly increased the binary size and led to hitting the PyPI repository limit. With the new release, we have made a significant improvement. The new release can now dynamically load NCCL from an external source, reducing the binary size. For the PyPI package, the ``nvidia-nccl-cu12`` package will be fetched during installation. With more downstream packages reusing NCCL, we expect the user environments to be slimmer in the future as well. (#9796, #9804, #10447)
 
+***************************************************
+Parts of the Python package now require glibc 2.28+
+***************************************************
+Starting from 2.1.0, XGBoost Python package will be distributed in two variants:
+
+* ``manylinux_2_28``: for recent Linux distros with glibc 2.28 or newer. This variant comes with all features enabled.
+* ``manylinux2014``: for old Linux distros with glibc older than 2.28. This variant does not support GPU algorithms or federated learning.
+
+The ``pip`` package manager will automatically choose the correct variant depending on your system.
```

Comment:
I ran some tests locally to verify that `pip` picks up the correct variant, depending on the glibc version on the system.